core/src/main/scala/org/apache/spark/sql/collection/Utils.scala

```diff
@@ -575,6 +575,39 @@ object Utils {
     SQLExecution.withNewExecutionId(ctx, queryExecution)(body)
   }
 
+  def immutableMap[A, B](m: mutable.Map[A, B]): Map[A, B] = new Map[A, B] {
```

Comment:
Just to understand. Why these '+' and '-' needed to be overridden and defaults were not ok for immutable map.